pinot-core/src/main/java/org/apache/pinot/core/util/SegmentProcessorAvroUtils.java

```diff
@@ -130,4 +180,41 @@ public static Schema convertPinotSchemaToAvroSchema(org.apache.pinot.spi.data.Sc
     }
     return fieldAssembler.endRecord();
   }
+
```

Comment:
Is there a reason for using custom conversion versus org.apache.avro.DecimalConversion ?

Reply:
More or less is what I said in https://github.com/apache/pinot/pull/16210/files#r2200793757.